src/session/sidebar/row.rs

```diff
@@ -72,223 +73,255 @@ mod imp {
+                            }
+                        }
+                    },
+                );
+                let unread_count_css_expression = gtk::ClosureExpression::new(
+                    |args| {
+                        let notification_settings =
+                            args[1].get::<BoxedChatNotificationSettings>().unwrap().0;
+                        let scope_notification_settings =
+                            args[2].get::<BoxedScopeNotificationSettings>().unwrap().0;
+
+                        vec![
+                            "unread-count".to_string(),
+                            if notification_settings.use_default_mute_for {
+                                if scope_notification_settings
+                                    .map(|s| s.mute_for > 0)
+                                    .unwrap_or(notification_settings.mute_for > 0)
+                                {
+                                    "unread-count-muted"
+                                } else {
+                                    "unread-count-unmuted"
+                                }
+                            } else if notification_settings.mute_for > 0 {
                                 "unread-count-muted"
                             } else {
                                 "unread-count-unmuted"
                             }
-                        } else if notification_settings.mute_for > 0 {
-                            "unread-count-muted"
-                        } else {
-                            "unread-count-unmuted"
-                        }
-                        .to_string(),
-                    ]
-                },
-                &[
-                    notification_settings_expression.upcast(),
-                    scope_notification_settings_expression.upcast(),
-                ],
-            );
-            unread_count_label_css_expression.bind(
-                &*self_.unread_count_label,
-                "css-classes",
-                gtk::NONE_WIDGET,
-            );
+                            .to_string(),
+                        ]
+                    },
+                    &[
+                        notification_settings_expression.upcast(),
+                        scope_notification_settings_expression.upcast(),
+                    ],
+                );
+                unread_count_expression.bind(&*self_.unread_count_label, "label", Some(chat));
+                unread_count_visibility_expression.bind(
+                    &*self_.unread_count_label,
+                    "visible",
+                    Some(chat),
+                );
+                unread_count_css_expression.bind(
+                    &*self_.unread_count_label,
+                    "css-classes",
+                    Some(chat),
+                );
+
+                // Pin icon bindings
+                let pin_visibility_expression = gtk::ClosureExpression::new(
+                    move |args| {
```

Comment:
Not part of the PR, but as we do massive refactorings anyway: This `move` is not needed.